### PR TITLE
Lint tools in CI

### LINKED ISSUE #917: Run cargo fmt in CI
I have removed all further references to Bors in the repository. I ran `cargo fmt`, but didn't realize it would mess up the diff for `migration_guide.rs` and `helpers.rs`. Tell me if you need me to do anything else! :)

_Originally posted by @BD103 in https://github.com/bevyengine/bevy-website/issues/912#issuecomment-1927423101_

### PR DESCRIPTION
**Pull request by BD103**
Closes #917.

This PR runs Clippy and Rustfmt in CI in order to check all of the tools for proper formatting and catch bad code.

I additionally renamed the `test-code` job to `test-code-examples`, since I believe it better describes what the action does.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -48,7 +112,7 @@ jobs:
           retention-days: 1
 
   generate-errors:
-    needs: [markdownlint, test-code]
+    needs: [markdownlint, test-code-examples, lint-tools]
```
Comment (BD103) [inline .github/workflows/ci.yml:115]: When #923 is merged, `check-hide-lines` will also need to be added here.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -144,37 +208,9 @@ jobs:
...
-      - name: Install alsa and udev
-        run: sudo apt-get update; sudo apt-get install --no-install-recommends libasound2-dev libudev-dev
-
-      - name: Build & run doc tests
-        run: cd code-validation && cargo test
-
   build-website:
     runs-on: ubuntu-latest
-    needs: [markdownlint, generate-assets, generate-errors, generate-wasm-examples, generate-community]
+    needs: [markdownlint, test-code-examples, lint-tools, generate-assets, generate-errors, generate-wasm-examples, generate-community]
```
Comment (BD103) [inline .github/workflows/ci.yml:213]: Ditto about #923.

File: generate-assets/src/lib.rs [hunk 3]
```diff
@@ -511,36 +512,36 @@ fn get_bevy_version_from_manifest(
...
-    while !dependency.is_none() && !bevy_crate.is_none() {
+    while dependency.is_some() && bevy_crate.is_some() {
         let dependency_name = dependency.unwrap().0;
         let bevy_crate_name = bevy_crate.unwrap();
 
-        if dependency_name < bevy_crate_name {
-            dependency = dependencies.next();
-        } else {
-            if dependency_name == bevy_crate_name {
+        match dependency_name.cmp(bevy_crate_name) {
```
Comment (BD103) [inline generate-assets/src/lib.rs:526]: This was suggested in a lint, but I'm unsure if I like it as compared to the original. Thoughts?

File: generate-assets/src/lib.rs [hunk 4]
```diff
@@ -649,6 +655,8 @@ fn get_bevy_crates(db: &CratesIoDb) -> Result<Vec<(String, String)>, rusqlite::E
                 AND repository = ?\
         ",
     )?;
+
+    // Required let and return due to bevy_crates_statement not living long enough.
```
Comment (BD103) [inline generate-assets/src/lib.rs:659]: Is there a better solution that can keep `bevy_crates_statement` alive long enough?

File: sass/pages/_migration_guide.scss
Comment (BD103) [inline sass/pages/_migration_guide.scss:1]: This is a follow up from #939, where this file was not caught.

Nothing about this file has been changed except for the line endings.

**Review by AxiomaticSemantics — COMMENTED** [review]
This looks good to me, the linting itself leads to most of the changes here.

**Review by alice-i-cecile — APPROVED** [review]

**Pull request merged**